packages/trimerge-sync/src/TrimergeClient.test.ts

```diff
@@ -163,10 +163,35 @@ Object {
     );
     await timeout();
   });
+
   it('fails on leader event with no leader', async () => {
     const { onEvent } = makeTrimergeClient();
     // This just logs a warning, added for code coverage
     onEvent({ type: 'leader', clientId: '', action: 'accept' }, false);
     await timeout();
   });
+
+  it('preserves object references from client', async () => {
+    const { client } = makeTrimergeClient();
+
+    const nestedObject = {
+      field: 'value',
+    };
+
+    const doc = {
+      nested: nestedObject,
+    };
+
+    client.updateDoc(doc, 'message');
+
+    nestedObject.field = 'newValue';
```

Comment:
We should write this unit test without mutating any of the objects (i.e., treat them all as readonly). Mutability adds a pretty significant complication here that is harder to reason about.